src/main/java/org/apache/onami/persist/TransactionalAnnotationReader.java

```diff
@@ -60,7 +60,7 @@ Transactional readAnnotationFrom(MethodInvocation methodInvocation) {
   /**
    * Helper class for obtaining the default of {@link Transactional @Transactional}.
    */
-  @Transactional
+  @Transactional(onUnits = {})
```

Comment:
See above. if there is no annotation on a method invocation a default transactional is added.